keyboards/4pplet/waffling60/rev_a/rev_a.c

```diff
@@ -0,0 +1,84 @@
+#include "rev_a.h"
+
+void keyboard_pre_init_kb(void) {
+  // Call the keyboard pre init code.
+  init_led();
+}
+
+void init_led(void){
+  // Set on match, clear on TOP
+  TCCR1A  = ((1 << COM1A1) | (1 << COM1A0) | (1 << COM1B1)| (1 << COM1B0)| (1 << COM1C1)| (1 << COM1C0)); // all leds on.
+  // Phase + Frequency Correct PWM, Fcpu speed
+  TCCR1B  = ((1 << CS10) | (1 << WGM13));
+  // define end of timer
+  ICR1 = TIMER_TOP;
+  // default set to lowest, will not toggle
+  led_set_color(RGB_OFF);
+
+  brightness_level = DEFAULT_BRIGHTNESS_LEVEL;
+  // Set our LED pins as output
+  setPinOutput(B7);
+  setPinOutput(C5);
+  setPinOutput(C6);
+}
+
+
+layer_state_t layer_state_set_kb(layer_state_t state) { // just to illustrate, make changes if needed
+  if(LAYER_INDICATOR){
+    switch (get_highest_layer(state)) {
+    case _FN1:
+        led_set_color(RGB_PINK);
+        break;
+    case _FN2:
+        led_set_color(RGB_ORANGE);
+        break;
+    default:
+        led_set_color(RGB_OFF);
+        break;
+    }
+  }
+  return state;
+}
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res && LOCK_INDICATORS) {
+        // do not indicate multiple lock statuses in current implementation,
+        // sorted by most likely first (if capslock is on, it will be purple, regardless of numlock state)
+        if(led_state.caps_lock)
+          led_set_color(RGB_ORANGE);
+        else if(led_state.num_lock)
+          led_set_color(RGB_PINK);
+        else if(led_state.scroll_lock)
+          led_set_color(RGB_CYAN);
+        else if(led_state.compose)
+          led_set_color(RGB_PURPLE);
+        else if(led_state.kana)
+          led_set_color(RGB_YELLOW);
+        else
+          led_set_color(RGB_OFF);
+    }
+    return res;
+}
+
+void led_set_color(uint8_t r, uint8_t g, uint8_t b){
```

Comment:
Along the same lines a noroads, I would like to see this refactored to use rgblight custom driver so we get a consistent interface and can avoid things like the duplicate rgb colours within `keyboards/4pplet/waffling60/rev_a/rev_a.h`.